feat!: refactor into Component Module [TWO-548]

jai opened the pull request:
See https://www.notion.so/honestbank/WIP-How-to-structure-a-Terraform-module-31374a1594f84ef7b185ef4e06b36619

<!--
* Please link to at least one issue in the issue tracker. If an issue doesn't exist,
  create one.
* PR titles should follow https://www.conventionalcommits.org.
-->

### Info Required for All Commits

Description:

* Refactors this into a Component Module (see our [Terraform module design](https://www.notion.so/honestbank/WIP-How-to-structure-a-Terraform-module-31374a1594f84ef7b185ef4e06b36619)) to be used by https://github.com/honestbank/kubeflow-infrastructure
* Adds VPC and subnet tagging (closes https://github.com/honestbank/argoflow-aws-infrastructure/issues/3)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/honestbank/argoflow-aws-infrastructure/4)
<!-- Reviewable:end -->

jai committed:
feat!: refactor into Component Module

See https://www.notion.so/honestbank/WIP-How-to-structure-a-Terraform-module-31374a1594f84ef7b185ef4e06b36619

jai committed:
ci(terraform): update args

jai committed:
ci(terraform): copy contents from terraform-aws-eks

pgcloud reviewed (approved):
All good. Just wanted to call out the Terraform Cloud (login) action failing in github actions.

Once this is resolved, this LGTM

> Reviewed 5 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/honestbank/argoflow-aws-infrastructure/4#-:-Mjld3ZK5XGAAaW7MwSL:b-pxrk31)* status: 5 of 7 files reviewed, all discussions resolved (waiting on @theerasan)



<!-- Sent from Reviewable.io -->

theerasan reviewed (approved):
> Reviewed 2 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/honestbank/argoflow-aws-infrastructure/4)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jai)



<!-- Sent from Reviewable.io -->

jai committed:
feat!: add VPC and subnet tagging for EKS

BREAKING CHANGE: This change adds a new input variable `aws_vpc_id` in order to perform tagging. Not providing this input variable will result in a failed run.

jai committed:
docs(test): add readme

jai committed:
fix: tfsec warnings

jai committed:
docs(core): add readme

jai committed:
docs(readme): add prerequisites

jai committed:
feat(wrapper): add inputs and outputs

jai committed:
ci(pre-commit): enable tfsec
